File: evals/refs/gf.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func parseByte(s string) int {
	if strings.HasPrefix(s, "0x") || strings.HasPrefix(s, "0X") {
		v, _ := strconv.ParseInt(s[2:], 16, 64)
		return int(v)
	}
	v, _ := strconv.Atoi(s)
	return v
}

func gfmul(a, b int) int {
	p := 0
	for i := 0; i < 8; i++ {
		if b&1 != 0 {
			p ^= a
		}
		b >>= 1
		hi := a & 0x80
		a = (a << 1) & 0xFF
		if hi != 0 {
			a ^= 0x1B
		}
	}
	return p
}

func main() {
	in := bufio.NewScanner(os.Stdin)
	in.Buffer(make([]byte, 1<<20), 1<<20)
	in.Split(bufio.ScanWords)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()
	read := func() string { in.Scan(); return in.Text() }
	T, _ := strconv.Atoi(read())
	for c := 0; c < T; c++ {
		op := read()
		a := parseByte(read())
		b := parseByte(read())
		if op == "+" {
			fmt.Fprintln(out, a^b)
		} else {
			fmt.Fprintln(out, gfmul(a, b))
		}
	}
}

File: evals/refs/gf.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

static int parse_byte(const char *s) {
    if (s[0] == '0' && (s[1] == 'x' || s[1] == 'X')) {
        return (int)strtol(s+2, NULL, 16);
    }
    return atoi(s);
}

static int gfmul(int a, int b) {
    int p = 0;
    for (int i = 0; i < 8; i++) {
        if (b & 1) p ^= a;
        b >>= 1;
        int hi = a & 0x80;
        a = (a << 1) & 0xFF;
        if (hi) a ^= 0x1B;
    }
    return p;
}

int main(void) {
    int T;
    if (scanf("%d", &T) != 1) return 1;
    char op[8], sa[16], sb[16];
    while (T--) {
        if (scanf("%7s %15s %15s", op, sa, sb) != 3) return 1;
        int a = parse_byte(sa);
        int b = parse_byte(sb);
        if (op[0] == '+') printf("%d\n", a ^ b);
        else printf("%d\n", gfmul(a, b));
    }
    return 0;
}
